TELCODOCS-2170 - Adding docs for the T-GM antenna delay settings (#87860)
openshift/openshift-docs: modules/nw-ptp-e810-hardware-configuration-reference.adoc

====

Set `spec.profile.plugins.e810.ublxCmds` parameters to configure the GNSS clock in the `PtpConfig` custom resource (CR).

[IMPORTANT]
====

> You must configure separate delay offset values to counteract 1PPS signal delay and T-GM antenna cable signal delay.

Reviewer: ```suggestion
You must configure separate delay values to counteract 1PPS signal delay and T-GM antenna cable signal delay.
```

To calculate optimal offset values that do not introduce constant time errors, make precise measurements of the 1PPS cable and GNSS antenna cable signal delays.
Red Hat cannot assist in this measurement or provide any values for the required delay offsets.

> ====

Reviewer: A bit confused about our use of delay and offset together and interchangeably through the text. Can we use "offset" when we mean an adverse effect and "delay compensation" when we mean fixing  it?

> 
> [IMPORTANT]
> ====
> You must configure separate delay offset values to counteract 1PPS signal delay and T-GM antenna cable signal delay.
> To calculate optimal offset values that do not introduce constant time errors, make precise measurements of the 1PPS cable and GNSS antenna cable signal delays.
> Red Hat cannot assist in this measurement or provide any values for the required delay offsets.
> ====
> 

Reviewer: ```suggestion

[IMPORTANT]
====
You must configure separate delay values to counteract 1PPS signal delay and T-GM antenna cable signal delay.
To calculate optimal offset values that do not introduce constant time errors, make precise measurements of the 1PPS cable and GNSS antenna cable signal delays.
Red Hat cannot assist in this measurement or provide any values for the required delay offsets.
====

```

Each of these `ublxCmds` stanzas correspond to a configuration that is applied to the host NIC by using `ubxtool` commands.
For example:

[source,yaml]
----
ublxCmds:
  - args:
      - "-P"
      - "29.20"
      - "-z"
      - "CFG-HW-ANT_CFG_VOLTCTRL,1"
      - "-z"
      - "CFG-TP-ANT_CABLEDELAY,<calculated_delay_offset>"<1>
    reportOutput: false
----

> <1> Calculated T-GM antenna delay offset in microseconds, for example `25`.

Reviewer: ```suggestion
<1> Pre-calibrated T-GM antenna delay offset in nanoseconds, for example `25`.
```

Reviewer: Pre-calibrated or pre-measured, or something of sorts. The process they do for the delays must contain measurement anr / or calibration

====

Set `spec.profile.plugins.e810.ublxCmds` parameters to configure the GNSS clock in the `PtpConfig` custom resource (CR).

[IMPORTANT]
====
You must configure an offset value to compensate for T-GM GPS antenna cable signal delay.
To configure the optimal T-GM antenna offset value, make precise measurements of the GNSS antenna cable signal delay.

> Red Hat cannot assist in this measurement or provide any values for the required delay offsets.

Reviewer 2: ```suggestion
Red{nbsp}Hat cannot assist in this measurement or provide any values for the required delay offsets.
```

https://redhat-documentation.github.io/supplementary-style-guide/#non-breaking-spaces

> Use a non-breaking space ({nbsp}) between the words "Red" and "Hat". The non-breaking space prevents an automatic line break from separating the two words onto two lines.

====

Each of these `ublxCmds` stanzas correspond to a configuration that is applied to the host NIC by using `ubxtool` commands.
For example:

[source,yaml]
----
ublxCmds:
  - args:
      - "-P"
      - "29.20"
      - "-z"
      - "CFG-HW-ANT_CFG_VOLTCTRL,1"
      - "-z"
      - "CFG-TP-ANT_CABLEDELAY,<antenna_delay_offset>"<1>
    reportOutput: false
----

> <1> Measured T-GM antenna delay offset in nanoseconds.
> To get the required delay offset value, you must measure the cable delay using external test equipment.

Reviewer 2: ```suggestion
<1> Measured T-GM antenna delay offset in nanoseconds. To get the required delay offset value, you must measure the cable delay using external test equipment.
```

- To avoid hard wrapping as per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-file-guidelines

Author: One sentence per line is not "hard wrapping". See https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line
